jwst/ramp_fitting/utils.py

```diff
@@ -401,39 +451,32 @@ def alloc_arrays(n_int, imshape, max_seg):
 
     segs_4: integer, 4D array
         Hypercube of lengths of segments for all integrations and pixels
-
-    num_seg_per_int: integer, 3D array
-        Cube of numbers of segments for all integrations and pixels
     """
-    slope_int = np.zeros((n_int,) + imshape, dtype=np.float32)
-    dq_int = np.zeros((n_int,) + imshape, dtype=np.uint32)
-
-    # for integration-specific variances
-    var_p3 = np.zeros((n_int,) + imshape, dtype=np.float32)
-    var_r3 = var_p3.copy() * 0.
-    var_both3 = var_r3.copy() * 0.
+    # Initialize variances so that non-existing ramps and segments will have
+    #   negligible contributions
+    # Integration-specific:
+    var_p3 = np.zeros((n_int,) + imshape, dtype=np.float32) + LARGE_VARIANCE
+    var_r3 = var_p3.copy()
+    var_both3 = var_p3.copy()
     s_inv_var_p3 = var_p3.copy() * 0.
```

Comment:
Instead of copying and multiplying, you could use `s_inv_var_p3 = np.zeros_like(var_p3)`.  I doubt that that would make any noticeable difference, but in principle it should be quicker and use less temporary memory.